ESP8266/ESP8266.cpp

```diff
@@ -16,14 +16,22 @@
 
 #include "ESP8266.h"
 
+#define ATPARSER_BUFSIZE    256  //bytes
+#define ATPARSER_TIMEOUT    8000 //ms
+#define BAUDRATE            115200
+
 ESP8266::ESP8266(PinName tx, PinName rx, bool debug)
-    : _serial(tx, rx, 1024), _parser(_serial)
-    , _packets(0), _packets_end(&_packets)
+    :  _packets(0), _packets_end(&_packets)
 {
-    _serial.baud(115200);
-    _parser.debugOn(debug);
+    _fh = new BufferedSerial(tx, rx, BAUDRATE);
+    _parser =  new ATParser(*_fh, ATPARSER_BUFSIZE, ATPARSER_TIMEOUT, debug);
```

Comment:
There will be cases where this WifiInterface will be used in conjunction with other XYZInterfaces .. In those scenarios, it will be preferable to shutdown something and completely get rid of all memory allocations relating to that entity .. I'm thinking about multi radio systems , wifi-cellular-BT all togather. An interface does its job and then gets shutdown and makes space for other processes to run. For a standalone wifi example your solution is great, but we should also be thinking about real life applications and use cases where indeed lifetime of this interface might cut short.